### ISSUE TITLE
[Feature Request] Add accumulated api requests  made to cloudwatch API endpoint

### ISSUE THREAD

**Issue by jnmarquis** [influxdata/telegraf#5199]
As CloudWatch is not free beyond the 1M requests, telegraf should keep a counter of the api requests made to CloudWatch API in order to estimate monthly costs for this service.
cloudwatch_exporter from Prometheus provides this counter and should not be very difficult to implement in telegraf.

**Comment by danielnelson** [comment]
Sounds like a good idea, is this something you are interested in working on?

**Comment by jnmarquis** [comment]
I'm no Go expert, but could give it a go

**Comment by danielnelson** [comment]
Yeah give it a shot, it might make sense to do it as a `selfstat.Stat`.  There is an example of how to use it in [influxdb_listener](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/influxdb_listener/http_listener.go#L63).  These metrics are only emitted when the `internal` input plugin is enabled.

**Comment by sspaink** [comment]
@jnmarquis seems tracking API requests hasn't been implemented yet, but it looks like https://github.com/influxdata/telegraf/pull/5544 helped make the cloudwatch plugin more cost effective which could help. Are you still interested in seeing feature implemented?
